### PR TITLE
Modules everywhere

### PR DESCRIPTION
**Pull request by ascott1**
This moves the existing code into what I think are sensible modules and adds tests for (nearly) everything. I'll add tests for the remainder for the Unsubsidized Stafford loan logic next week.

Current test coverage:

```
---------------------------|----------|----------|----------|----------|----------------|
File                       |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
---------------------------|----------|----------|----------|----------|----------------|
 src/                      |      100 |      100 |      100 |      100 |                |
  calc-debt.js             |      100 |      100 |      100 |      100 |                |
  cost.js                  |      100 |      100 |      100 |      100 |                |
  debt-total.js            |      100 |      100 |      100 |      100 |                |
  default-values.js        |      100 |      100 |      100 |      100 |                |
  index.js                 |      100 |      100 |      100 |      100 |                |
  payment.js               |      100 |      100 |      100 |      100 |                |
  rates.js                 |      100 |      100 |      100 |      100 |                |
  scholarship.js           |      100 |      100 |      100 |      100 |                |
  year-one.js              |      100 |      100 |      100 |      100 |                |
 src/loans/                |    79.05 |    55.26 |      100 |    79.05 |                |
  grad-plus.js             |      100 |      100 |      100 |      100 |                |
  index.js                 |      100 |      100 |      100 |      100 |                |
  institution.js           |      100 |      100 |      100 |      100 |                |
  perkins.js               |      100 |      100 |      100 |      100 |                |
  private.js               |      100 |      100 |      100 |      100 |                |
  subsidized-stafford.js   |      100 |      100 |      100 |      100 |                |
  unsubsidized-stafford.js |    56.94 |    45.16 |      100 |    56.94 |... 162,165,166 |
 src/utils/                |      100 |      100 |      100 |      100 |                |
  enforce-range.js         |      100 |      100 |      100 |      100 |                |
  merge.js                 |      100 |      100 |      100 |      100 |                |
---------------------------|----------|----------|----------|----------|----------------|
All files                  |    87.84 |       66 |      100 |    87.84 |                |
---------------------------|----------|----------|----------|----------|----------------|
```
## Todo
- Finish Unsubsidized Stafford loan tests
- Gradually refactor now that tests are in place
## Review
- @higs4281

### CODE REVIEW

**Comment by mistergone** [comment]
This is great! Now I have a clear idea of how you wanted to modularize the code!

File: src/debt-total.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  // Grad Plus debt at graduation
+  data.gradplusTotal = calcDebt(
+    data.gradplusWithFee,
+    data.gradplusRate,
+    data.programLength,
+    data.deferPeriod
+  );
+
+  // Parent Plus Loans with origination fees
+  data.parentplusWithFee = data.parentplus * data.plusOriginationFee;
```
Comment (higs4281) [inline src/debt-total.js:57]: just wondering: 
i'm assuming plusOriginationFee is a percentage and parentplus is the loan amount.
if plusWithFee represents the loan amount including the fee, wouldn't you need to calculate it this way?

```
data.parentplusWithFee = data.parentplus + (data.parentplus * data.plusOriginationFee) 
```

Comment (ascott1) [inline src/debt-total.js:57]: Great question and that certainly makes sense to me. I haven't touched any of the calculations, just moved things around and cleaned them up. I've been collecting a list of similar questions for Patrick and I'll add this to the list.

**Comment by ascott1** [comment]
The last push increases the test coverage to nearly 100% (and as close as I can get it without discussing some logic bugs I've found with our student loan expert):

```
---------------------------|----------|----------|----------|----------|----------------|
File                       |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
---------------------------|----------|----------|----------|----------|----------------|
 src/                      |      100 |      100 |      100 |      100 |                |
  calc-debt.js             |      100 |      100 |      100 |      100 |                |
  cost.js                  |      100 |      100 |      100 |      100 |                |
  debt-total.js            |      100 |      100 |      100 |      100 |                |
  default-values.js        |      100 |      100 |      100 |      100 |                |
  index.js                 |      100 |      100 |      100 |      100 |                |
  payment.js               |      100 |      100 |      100 |      100 |                |
  rates.js                 |      100 |      100 |      100 |      100 |                |
  scholarship.js           |      100 |      100 |      100 |      100 |                |
  year-one.js              |      100 |      100 |      100 |      100 |                |
 src/loans/                |      100 |    89.06 |      100 |      100 |                |
  grad-plus.js             |      100 |      100 |      100 |      100 |                |
  index.js                 |      100 |      100 |      100 |      100 |                |
  institution.js           |      100 |      100 |      100 |      100 |                |
  perkins.js               |      100 |      100 |      100 |      100 |                |
  private.js               |      100 |      100 |      100 |      100 |                |
  subsidized-stafford.js   |      100 |      100 |      100 |      100 |                |
  unsubsidized-stafford.js |      100 |       86 |      100 |      100 |                |
 src/utils/                |      100 |      100 |      100 |      100 |                |
  enforce-range.js         |      100 |      100 |      100 |      100 |                |
  merge.js                 |      100 |      100 |      100 |      100 |                |
---------------------------|----------|----------|----------|----------|----------------|
All files                  |      100 |    92.05 |      100 |      100 |                |
---------------------------|----------|----------|----------|----------|----------------|
```

@higs4281 since this is a structural refactor and doesn't change any logic, I propose that we merge this in and I will discuss your above question (and some others) with Patrick and address them in a future pull request.

**Comment by higs4281** [comment]
@ascott1 agreed.

**Comment by ascott1** [comment]
Coolness. Merging!

**Pull request merged**